panel/io/state.py

```diff
@@ -379,8 +379,9 @@ def as_cached(self, key: str, fn: Callable[[], None], ttl: int = None, **kwargs)
         return ret
 
     def add_periodic_callback(
-        self, callback: Callable[[], None], period: int=500,
-        count: Optional[int] = None, timeout: int = None, start: bool=True
+        self, callback: Union[Callable[[], None], Coroutine[Any, Any, None]],
```

Comment:
I believe the preferred syntax for listing types in the code base is `|` instead of `Union`. Is that right @philippjfr ?

```suggestion
        self, callback: Callable[[], None] | Coroutine[Any, Any, None],
```

Reply:
No `|` is preferred.

Reply:
I didn't go for the pipe because I noticed that Python 3.6 is still supported in the code base. As far as I know that feature has been backported to 3.7 and above.

But I prefer it as well, 3.6 is EOL anyhow.

Reply:
In practice Panel has dropped support of Python 3.6 as it relies on Bokeh >= 2.4.0 which itself no longer supports it. The setup.py file still has to be updated to reflect that.

Reply:
@WoLpH the test suite is failing as `Coroutine` is not imported. Would you like to fix that?

Reply:
Oops... not sure how that was not committed but I only have that change locally. I've added it and updated using the `|` instead of `Union`